cloudbridge/cloud/providers/azure/azure_client.py

```diff
@@ -278,6 +290,39 @@ def create_storage_account(self, name, params):
         return self.storage_client.storage_accounts. \
             create(self.resource_group, name.lower(), params).result()
 
+    # Create a storage account. To prevent a race condition, try
+    # to get or create at least twice
+    @tenacity.retry(stop=tenacity.stop_after_attempt(2),
+                    retry=tenacity.retry_if_exception_type(CloudError),
+                    reraise=True)
+    def _get_or_create_storage_account(self):
+        if self._storage_account:
+            return self._storage_account
+        else:
+            try:
+                self._storage_account = \
+                    self.get_storage_account(self.storage_account)
+            except CloudError as cloud_error:
+                if "not found" in cloud_error.message:
```

Comment:
Is there something like an error code we could use here instead? Is the error message potentially localized, and therefore language dependent?

Reply:
They have status codes which are 404 for not found and 403 for permission denied, and error codes which are ResourceNotFound/ResourceGroupNotFound and AuthorizationFailed. I'm wary of using the status codes since a 404 could mean that the resource group doesn't exist or that the storage account doesn't exist when trying to get a storage account within a resource group, but I think we could use ResourceNotFound/ResourceGroupNotFound since there shouldn't be any other resource than the one we're getting that could cause this in that particular spot.